invenio_communities/assets/semantic-ui/js/invenio_communities/settings/curation-policy.js

```diff
@@ -0,0 +1,146 @@
+/*
+ * This file is part of Invenio.
+ * Copyright (C) 2023 CERN.
+ *
+ * Invenio is free software; you can redistribute it and/or modify it
+ * under the terms of the MIT License; see LICENSE file for more details.
+ */
+
+import { i18next } from "@translations/invenio_communities/i18next";
+import { Formik } from "formik";
+import _defaultsDeep from "lodash/defaultsDeep";
+import React, { Component } from "react";
+import ReactDOM from "react-dom";
+import { FieldLabel, RichInputField, CKEditorConfig } from "react-invenio-forms";
+import { Button, Form, Grid, Icon, Message } from "semantic-ui-react";
+import { CommunityApi } from "../api";
+import { communityErrorSerializer } from "../api/serializers";
+import PropTypes from "prop-types";
+import * as Yup from "yup";
+
+const COMMUNITY_VALIDATION_SCHEMA = Yup.object({
```

Comment:
I think for small forms, it's actually nice to have client-side validation and quick feedback (vs. having to always submit to the backend to get the validation errors). IIRC Formik also has baked-in support for Yup, and that's why we use it.

The deposit form is a beast on its own, so I think the "slow" backend-based feedback is fine (and also allows the form to be "lighter").

So I'm pro-Yup, but would be fine also if we decided to skip it.

Reply:
I think the more centalized validation is better, we currently have:
- Marshmallow in the backend
- Very light on formik (?)

We would add yup on the frontend. I'm not against this if it improves the UX, however we should document in some _best practices_ section a clear cut of when is it used and when is it not. To avoid having inconsistencies.

Reply:
Yup is already used in `react-invenio-deposit`, `react-invenio-forms` and `invenio-communities` at least, see [here](https://github.com/search?q=org%3Ainveniosoftware+yup+extension%3A.js&type=Code&ref=advsearch&l=&l=).

Reply:
I have seen it, but I think it was maybe introduced without discussion. Maybe it would be better to keep validation consistent, as Pablo says, if not documented it might diverge even more